### ISSUE TITLE
Have access to field-level props in validate function

### ISSUE THREAD

**Issue by funwithtriangles** [erikras/redux-form#2760]
It would be nice to have access to field level props when validating. For instance, when trying to get two passwords to match, you could do this:

```html
<Field name='pass2' type='password' validate={matchesField} matchWith='pass1' />
```
And then...
```javascript
export const matchesField = (value, allValues, formProps, fieldProps) =>
  value !== allValues[fieldProps.matchWith] ? 'matchesField' : undefined
```
Would this not be quite useful? :)

**Comment by funwithtriangles** [comment]
Ah, I see you can do this:

```html
<Field name='pass2' type='password' validate={matchesField('pass1')}  />
```
And then...
```javascript
export const matchesField = (matchWith) => (value, allValues, formProps, fieldProps) =>
  value !== allValues[matchWith] ? 'matchesField' : undefined
```

Although I'm not sure if calling the validate function instead of passing it is good practice? It seems to fire onMount...

**Comment by gustavohenke** [comment]
This is already possible. You just had to look at the docs for `validate` prop!

http://redux-form.com/6.6.1/docs/api/Field.md/#-validate-value-allvalues-props-error-optional-

**Comment by funwithtriangles** [comment]
I did look at that exact piece of documentation and I don't see anything related to fieldProps. I'm proposing a fourth argument. My second example does work with the current API but it isn't very testable passing in a called function like that.

**Comment by gustavohenke** [comment]
OK, I misread your OP. Anyway, will leave this closed in favor of #2745.
